### ISSUE TITLE
Singular Matrix Error

### ISSUE THREAD

**Issue by shihabkhan1** [GeoStat-Framework/PyKrige#87]
I'm trying to carry out one dimensional kriging using the python module 'pykrige' but I'm getting a few errors.

I tried to pass a zero vector as argument for the second variable.

My code is as below

            hs = np.array(np.arange(0,1,bw))
            hs0 = np.zeros(len(hs))

            OK = OrdinaryKriging(P[:, 0], P[:, 1], P[:, 2], variogram_model='linear',
                     verbose=False, enable_plotting=False)
            
            z, ss = OK.execute('grid', hs, hs0)

It gives me a linear algebra error when I run the `execute` command.

    
      File "/home/home_shihab/miniconda3/lib/python3.6/site-packages/pykrige/ok.py", line 396, in _exec_vector
        a_inv = scipy.linalg.inv(a)
    
      File "/home/home_shihab/miniconda3/lib/python3.6/site-packages/scipy/linalg/basic.py", line 976, in inv
        raise LinAlgError("singular matrix")
    
    LinAlgError: singular matrix

Can anyone please help me with this?

**Comment by bsmurphy** [comment]
Sorry for the slow response. Have you tried using ``` 'points' ``` instead of ``` 'grid' ```? I suspect there's something going wrong with trying to meshgrid a 1D set of points...

**Comment by Yambottle** [comment]
@bsmurphy 
I met this error as well. Probably this could help to make it work, using another way to invert the cov-matrix. https://stackoverflow.com/a/54725420/12155679
@shihabkhan1 
I don't you still need this or not, please try to reduce or increase the dimension(rows) of your input. Good Luck!

**Comment by MuellerSeb** [comment]
We could implement the pseudo-inverse for the case where we have a singular matrix, that can occur frequently when the number of points is to low, or the variogram-model is ill-choosen.

**Comment by MuellerSeb** [comment]
I'll implement this in v1.5.1.

**Comment by MuellerSeb** [comment]
You could try PyKrige from this branch: https://github.com/GeoStat-Framework/PyKrige/tree/pseudo_inv

You can directly install it with pip:
```
pip install git+https://github.com/GeoStat-Framework/PyKrige.git@pseudo_inv
```

be sure to provide cython beforehand.

In this branch, the inverse is calculated by the pseudo-inverse.

**Comment by MuellerSeb** [comment]
This should be fixed with `v1.5.1` that will be released tomorrow.
Closing for now. Feel free to re-open, if you have further questions.
